File: backend/app/src/main/scala/com/github/ai/leetcodequiz/data/db/model/ProblemEntity.scala
package com.github.ai.leetcodequiz.data.db.model

final case class ProblemEntity(
  id: ProblemId,
  title: String,
  content: String,
  category: String,
  url: String,
  difficulty: String,
  likes: Int,
  dislikes: Int
)

File: android/app/src/main/java/com/aivanovski/leetcode/android/data/database/model/ProblemEntity.kt
package com.aivanovski.leetcode.android.data.database.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "problems")
data class ProblemEntity(
    @PrimaryKey(autoGenerate = false)
    @ColumnInfo("id")
    val id: Int,
    val title: String,
    @ColumnInfo("category_title")
    val categoryTitle: String,
    val difficulty: String,
    val url: String
)